thunder/torch/__init__.py

```diff
@@ -2009,6 +2009,26 @@ def tanhshrink(a: TensorLike, /) -> TensorLike:
 
 _inplace_to_out_of_place[tanhshrink] = tanhshrink, -1
 
+
+@torchsymbol(torch.threshold, torch.nn.functional.threshold, id="torch.threshold", is_method=False)
+def threshold(a: TensorProxy, /, threshold: float, value: float, inplace: bool = False) -> TensorLike:
```

Comment:
PyTorch reference implementation for comparison:

https://github.com/pytorch/pytorch/blob/781d28e2655f88ae2fef827ed110f22ed553a0ab/torch/_refs/nn/functional/__init__.py#L917

The PyTorch reference implementation uses a different comparison -- is the NaN propagation different between the two?

Reply:
Indeed it does not propagate NaNs correctly.  Changing.

Reply:
I was foolishly checking this on cpu tensors and bumping into https://github.com/Lightning-AI/lightning-thunder/issues/1936.